src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -860,34 +875,53 @@ private static GapicProductConfig createDummyInstance(
     return ImmutableMap.copyOf(mergedResourceNameConfigs);
   }
 
-  private static ImmutableMap<String, SingleResourceNameConfig> createSingleResourceNameConfigs(
+  private static void createSingleAndFixedResourceNameConfigs(
       DiagCollector diagCollector,
       ConfigProto configProto,
+      ImmutableMap.Builder<String, SingleResourceNameConfig> singleResourceNamesBuilder,
+      ImmutableMap.Builder<String, FixedResourceNameConfig> fixedResourceNamesBuilder,
       @Nullable List<ProtoFile> sourceProtos,
       TargetLanguage language) {
     ProtoFile file = null;
     if (sourceProtos != null) {
       file = sourceProtos.get(0);
     }
+    List<CollectionConfigProto> allCollectionConfigProtos =
+        new ArrayList<>(configProto.getCollectionsList());
+    configProto
+        .getInterfacesList()
+        .stream()
+        .forEach(i -> allCollectionConfigProtos.addAll(i.getCollectionsList()));
+
     LinkedHashMap<String, SingleResourceNameConfig> singleResourceNameConfigsMap =
         new LinkedHashMap<>();
-    for (CollectionConfigProto collectionConfigProto : configProto.getCollectionsList()) {
-      createSingleResourceNameConfig(
-          diagCollector, collectionConfigProto, singleResourceNameConfigsMap, file, language);
-    }
-    for (InterfaceConfigProto interfaceConfigProto : configProto.getInterfacesList()) {
-      for (CollectionConfigProto collectionConfigProto :
-          interfaceConfigProto.getCollectionsList()) {
+    for (CollectionConfigProto collectionConfigProto : allCollectionConfigProtos) {
+      PathTemplate pathTemplate = PathTemplate.create(collectionConfigProto.getNamePattern());
+      try {
+        pathTemplate.instantiate(new HashMap<>());
+      } catch (ValidationException e) {
+        // This must be a single resource name, because it has binding vars in the path pattern.
```

Comment:
I don't think I'm a fan of using exceptions like this if we don't need to. Why not check the name pattern for `{` or `*` chars?

Reply:
i suppose if we aren't terribly worried about accuracy, plus there's only one fixed resource name in existence
done.